packages/functionals/botpress-qna/src/views/index.js

```diff
@@ -344,18 +348,90 @@ export default class QnaAdmin extends Component {
+    formData.set('isReplace', this.state.isCsvUploadReplace)
+    formData.append('csv', this.state.csvToUpload)
+    const headers = { 'Content-Type': 'multipart/form-data' }
+    this.props.bp.axios
+      .post('/api/botpress-qna/csv', formData, { headers })
+      .then(() => {
+        this.setState({ importCsvModalShow: false })
+        this.fetchData()
+      })
+      .catch(({ response: { data: csvUploadError } }) => this.setState({ csvUploadError }))
+  }
+
   render() {
     return (
       <Panel>
         <Panel.Body>
-          <p>
+          <FormGroup>
+            <ButtonToolbar>
+              <Button
+                bsStyle="success"
+                onClick={() =>
+                  this.setState({
+                    importCsvModalShow: true,
+                    csvToUpload: null,
+                    csvUploadError: null,
+                    isCsvUploadReplace: false
+                  })}
+                type="button"
+              >
+                <Glyphicon glyph="upload" /> &nbsp; Import from CSV
+              </Button>
+              <Modal show={this.state.importCsvModalShow} onHide={() => this.setState({ importCsvModalShow: false })}>
+                <Modal.Header closeButton>
+                  <Modal.Title>Import CSV</Modal.Title>
+                </Modal.Header>
+                <Modal.Body>
+                  {this.state.csvUploadError && (
+                    <Alert bsStyle="danger" onDismiss={() => this.setState({ csvUploadError: null })}>
+                      <p>{this.state.csvUploadError}</p>
+                    </Alert>
+                  )}
+                  <form>
+                    <FormGroup>
+                      <ControlLabel>CSV-file</ControlLabel>
+                      <FormControl
+                        type="file"
+                        accept=".csv"
+                        onChange={e => this.setState({ csvToUpload: e.target.files[0] })}
+                      />
+                      <HelpBlock>CSV should be formatted &quot;question,answer_type,answer&quot;</HelpBlock>
+                    </FormGroup>
+                    <FormGroup>
+                      <Checkbox
+                        checked={this.state.isCsvUploadReplace}
+                        onChange={e => this.setState({ isCsvUploadReplace: e.target.checked })}
+                      >
+                        Replace existing FAQs
+                      </Checkbox>
+                      <HelpBlock>Deletes existing FAQs and then uploads new ones from the file</HelpBlock>
+                    </FormGroup>
+                  </form>
+                </Modal.Body>
+                <Modal.Footer>
+                  <Button bsStyle="primary" onClick={this.uploadCsv} disabled={!Boolean(this.state.csvToUpload)}>
+                    Upload
+                  </Button>
+                </Modal.Footer>
+              </Modal>
+
+              <Button bsStyle="success" href="/api/botpress-qna/csv" type="button">
```

Comment:
It's not protected by any way, right? 
Does this mean it's vulnerable to being downloaded by complete strangers just having the bot URL?

Reply:
@emirotin @epaminond Could you please check this? By default the `getRouter()` returns a secured router. Not sure with multer what the behavior is but we need to make sure this is secure

Reply:
I mean it's a plan link fetched over GET, how is it secured? It's not getting any headers

Reply:
Oh you're right. That most certainly doesn't work on a secured dashboard (NODE_ENV=prod). Need to use Axios.

Reply:
Hmmm, does this mean I've recently broken the logs archive download in a recent PR? It went mithe the breaking change so not that bad. Will check and fix today

Reply:
@emirotin , @slvnperron , I've tested it and this route _is_ secured: download only works if user is logged in as admin (the same way it works for other routes in this module).

Reply:
I wonder how it's possible. Unless we're using cookies there's no way to send secret with GET request

Reply:
@emirotin , actually you were right, this button always returned 401 status even when authorized. So I implemented a workaround to download it through axios! Nice catch!

Reply:
Now I need to go back and check the logs stuff I could presumably expose or break :)